cluster/saltbase/salt/node-upgrade-gcp/node-upgrade-gcp.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Monitors node configuration data from the metadata server and runs node
+# upgrades when they change.
+
+# Settings
+# POLL_INTERVAL is how frequently to issue curl requests after a failed curl.
+readonly POLL_INTERVAL=10
+# CACHE_DIR is where metadata files are saved once they've been curled.
+readonly CACHE_DIR="metadata-cache"
+# *_EXT denote the role of a saved metadata file.
+readonly CACHE_EXT="cache"
+readonly LATEST_EXT="latest"
+# WATCH_KEYS are the metadata keys for which, if any of their values change, a
+# node upgrade is performed.
+readonly WATCH_KEYS=("kube-env" "startup-script")
+
+# get-metadata curls all keys in WATCH_KEYS from the metadata server, retrying
+# until success, and saves them in CACHE_DIR named by the key and the provided
+# extension.
+#
+# Assumed vars:
+#   WATCH_KEYS
+#   CACHE_DIR
+#   POLL_INTERVAL
+#
+# Args:
+#  $1 -- extension to use for files (e.g. CACHE_EXT or LATEST_EXT)
+function get-metadata() {
+	local ext="${1}"
+	echo "$(date) Attempting to get metadata."
+	for k in "${WATCH_KEYS[@]}"; do
+		echo "$(date) Attempting to get metadata with key '${k}'."
+		local code=404
+		local cache_file="${CACHE_DIR}/${k}.${ext}"
+		until [[ ${code} -eq 200 ]]; do
```

Comment:
I don't understand the complexity of this loop, or bothering to explicity write the code. Is there any advantage to this over what we do elsewhere, which would just collapse this loop to:
```
until curl --fail --silent -H 'Metadata-Flavor: Google' "http://metadata/computeMetadata/v1/instance/attributes/${k}"; do sleep ${POLL_INTERVAL}; done
```
?